### DISCUSSION TITLE
Feature request: settle payments between searchers and miners on L2

### DISCUSSION THREAD

**Discussion by bertmiller** [flashbots/pm#64] [searchers]
Instead of searchers paying miners on L1 we could create a way for searchers and miners to settle payments on a L2 solution, such as Optimism.

The net effect of this would be to further reduce on-chain congestion.

From Georgios: "the miner would look at that state on the committed L1 contract" [to verify their payment]

I anticipate that this solution would need more research into the implementation details, as well as socialization with searchers and miners, but conceptually it seems sound and I wanted to record it.

**Comment by thegostep** [comment]
I have trouble understanding how this would work.

Bringing a payment from L2 to L1 requires a transaction on chain, which would need to be in a block after the flashbots bundle.

How would the payment be linked to the bundle?
